### PR TITLE
Python only builds

### PR DESCRIPTION
**Pull request by achilleas-k**
Setup script updated to support building without the NIX dependency.

The setup script detects if NIX is available by attempting to compile a dummy C++ file that simply imports `nix.hpp` and checking if it fails (see new file `checknix.py`).

Additionally, Python only builds can now be explicitly performed using `python setup.py build --pyonly`.

Travis build config now includes 4 configurations: Python 2.7 and 3.4 with and without NIX.

### CODE REVIEW

**Comment by achilleas-k** [comment]
Looks like it's not detecting NIX properly. Fixing now.

**Comment by achilleas-k** [comment]
Fixed.

**Comment by achilleas-k** [comment]
Relevant issues: #199, #200.

File: checknix.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    compiler.include_dirs.extend(incdirs)
+    distutils.sysconfig.customize_compiler(compiler)
+
+    stderr = os.dup(sys.stderr.fileno())
+    errfile = open(os.path.join(tmpdir, "check_nix.err"), 'w')
+    # redirect stderr to file
+    os.dup2(errfile.fileno(), sys.stderr.fileno())
+    try:
+        compiler.compile([file_name], output_dir=tmpdir,
+                         extra_postargs=["--std=c++11"])
```
Comment (gicmo) [inline checknix.py]: NIX (headers) also depend on boost (headers) so if boost is installed in a non-standard location then nix will fail to build. Could work around that by also passing the boost incdirs.

Comment (achilleas-k) [inline checknix.py]: I'll pass those as well.

**Pull request merged**